signer/src/transaction_signer.rs

```diff
@@ -723,17 +736,17 @@ where
             .into();
 
         match db.will_sign_bitcoin_tx_sighash(&sighash).await? {
-            Some(true) => Ok(()),
+            Some(true) => Ok(sighash),
             Some(false) => Err(Error::InvalidSigHash(sighash)),
```

Comment:
Is this the only reason for `will_sign = false`?

Reply:
Yeah. Maybe we should change the variant name, but the message for this error is "the given sighash is known and failed validation".